activemodel/lib/active_model/base.rb

```diff
@@ -0,0 +1,413 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  # = Active Model \Base
+  #
+  # Base provides subclasses with an Active Record-inspired
+  # interface to execute code with familiar methods like +.create+, +#save+, and
+  # +#update+. It includes the API module transitively through the Model module,
+  # so it's designed to integrate with Action Pack and Action View out of the box.
+  #
+  # Similar to the convention for applications to define an
+  # <tt>ApplicationRecord</tt> that inherits <tt>ActiveRecord::Base</tt>, it's
+  # also conventional for applications to define an <tt>ApplicationModel</tt>
+  # that inherits from Base:
+  #
+  #   # app/models/application_model.rb
+  #
+  #   class ApplicationModel < ActiveModel::Base
+  #   end
+  #
+  # Unlike other facets of Active Model, Base is a Class instead of a Module.
+  # Once classes have inherited from Base, they only need to define a +#save!+.
+  # For example, consider a <tt>Session</tt> model responsible for authenticating a
+  # <tt>User</tt>  with +email+ and +password+ credentials:
+  #
+  #   # app/models/session.rb
+  #
+  #   class Session < ApplicationModel
+  #     attr_accessor :email, :password, :request
+  #
+  #     validates :email, :password, presence: true
+  #
+  #     def save!
+  #       if (user = User.authenticate_by(email: email, password: password))
+  #         request.cookies[:signed_user_id] = user.signed_id
+  #       else
+  #         errors.add(:base, :invalid)
+  #
+  #         raise ActiveModel::ValidationError.new(self)
```

Comment:
I wonder if we should have an `ActiveModel::ModelInvalid` error (similar to ActiveRecord's.